src/backend/modules/company/company.service.ts

```diff
@@ -47,11 +53,35 @@ export class CompanyService {
           0,
         ),
         priceCents:
-          result.UserStock.flatMap((us) =>
-            us.SellOffer.map((so) => so.unitSellPriceCents),
-          ).reduce((all, cur) => (all < cur ? all : cur)) ?? null,
+          result.UserStock.length > 0
+            ? Math.min(
+              ...result.UserStock.flatMap((us) => 
+                us.SellOffer.map((so) => so.unitSellPriceCents),
+              ),
+            )
+          : 0 // I changed null to 0. I don't know if null was essential. Everything seems to work fine.
+        // priceCents:
+        //   result.UserStock.flatMap((us) =>
+        //     us.SellOffer.map((so) => so.unitSellPriceCents),
+        //   ).reduce((all, cur) => (all < cur ? all : cur)) ?? null,
       };
     });
+
+    if (orderBy != "") {
+      companies.sort((a,b) => {
+        const sortOrder = query.orderType === 'asc' ? 1: -1;
+        let result;
+        if(orderBy === "priceCents"){
+          result = (a.priceCents < b.priceCents) ? -1 : (a.priceCents > b.priceCents) ? 1 : 0;
+        } else {
+          result = (a.quantity < b.quantity) ? -1 : (a.quantity > b.quantity) ? 1 : 0;
+        }
```

Comment:
If there will be more variants to "orderBy" maybe a helper function is in store.
Something like
```TS
function compare(a: number, b:number) {
  return a > b ? 1 ? a < b : -1 : 0;
}
```